### ISSUE TITLE
Crashes after this error...

### ISSUE THREAD

**Issue by RCTORONTO** [xmrig/xmrig-cuda#89]
nvidia   thread #1 failed with error <randomx_prepare>:36 "invalid argument"
[2021-03-19 13:23:02.861]  nvidia   thread #0 failed with error <randomx_prepare>:36 "invalid argument"

I can see in nvidia-smi that the xmrig process stops using memory on my card, the program doesn't recover...

NVIDIA-SMI 460.56       Driver Version: 460.56       CUDA Version: 11.2
xmrig --version
XMRig 6.10.0
 built on Mar 16 2021 with GCC 7.5.0
 features: 64-bit AES

libuv/1.18.0
OpenSSL/1.1.1
hwloc/1

GeForce GTX 1050 
GeForce GT 710

testing Monero with older cards...

**Comment by Spudz76** [comment]
GT 710 is a Kepler.  Probably arch 35.
```
../xmrig-cuda/cmake/CUDA.cmake:    "   8.x for Fermi/Kepler          /Maxwell/Pascal,\n"
../xmrig-cuda/cmake/CUDA.cmake:    "   9.x for       Kepler          /Maxwell/Pascal/Volta,\n"
../xmrig-cuda/cmake/CUDA.cmake:    "  10.x for       Kepler          /Maxwell/Pascal/Volta/Turing,\n"
../xmrig-cuda/cmake/CUDA.cmake:    "  11.x for       Kepler (in part)/Maxwell/Pascal/Volta/Turing/Ampere\n\n"
../xmrig-cuda/cmake/CUDA.cmake:# Kepler GPUs are only supported with CUDA < 11.0
../xmrig-cuda/cmake/CUDA.cmake:                                "Use CUDA v10.x maximum, Kepler (30) was dropped at v11.")
```

Keplers only work with up to CUDA 10 Toolkit.

The 1050 is Pascal (probably arch 61) and will work with almost any combo of driver and CUDA.

Some newer drivers or newer toolkits drop support for old cards, running the latest may only work with the Pascal based card.  Best to match CUDA Toolkit version with driver version -- note your very new driver has CUDA 11.2 runtime, which would have to run in compatibility mode assuming that works right with apps compiled for Toolkit 10.2 (most of the time they do BUT...)

The driver readme/release-notes show what CUDA Runtime version is in the driver.

And then after all that, RandomX isn't really for GPUs.  Better to mine Haven or CN-GPU on a pool that pays out in XMR (like MoneroOcean) if you want to profit in XMR from running GPUs.  Algo was designed to never be useful on GPUs to make sure CPUs never become useless.

**Comment by RCTORONTO** [comment]
Its a shame because it does successfully work to chime in with the occasional accepted share the 1050 get's about 120 h/s on top of the 4300 h/s I get cpu mining on my xeons... I gave up on the 710 altogether it was only doing 20 h/s or so, I would add a few more GPU cards for extra finds that it is doing...(I have a pci-e backplane with lot's of extra slots already - and a "miner" usb type extension to use aswell.. - I wish I knew more about what it's doing and the cuda api so I could try and fix this... ah well

**Comment by Spudz76** [comment]
Note you're getting 120 hashes for 75 watts (1.6 hash-per-watt)

The CPU is getting 4300 for probably 180 watts of CPUs (23.8 hash-per-watt)

That GPU could score equivalent of ~13h/s if you mine CN-Heavy against an autochanging pool that pays in XMR.

**Comment by RCTORONTO** [comment]
I'm new to mining, I will look into pools offering cryptonight-heavy, I don't think the one I'm using does... (nicehash)

**Comment by Spudz76** [comment]
It's well worth it, here are my scores (in XMR equivalent hashrate) from MoneroOcean for a GTX1060 (`gm99-nv`) and a GTX970 (`gm02-nv`) which each score less than a CPU (maybe 620H/s on the GTX1060) directly running XMR (rx/0).

![image](https://user-images.githubusercontent.com/2391234/112684268-62b0d180-8e38-11eb-9720-e632ef3ecc95.png)

**Comment by RCTORONTO** [comment]
thanks for the cn-heavy tip - I got setup with a vault address and am mining xhv on miner.rocks now.... the hash rates have improved - I was getting about 130 h/s doing monero on the 1050 in xmrig with the cuda plugin (and it wasnt using the full ram when it was using it - I keep an eye on nvidia-smi).  The 710 was getting 20 h/s on monero.   now with both active I'm getting 310 h/s !! woo!   anyways this is all very interesting.... 

The other thing I have noticed is that the randomx prepare error I was getting mining monero on nice hash doesn't happen when I mine monero using the same script on minexmr ... strange it must be something to do with how their version of the randomx pooling works (with difficulty changing? I dunno)

**Comment by Spudz76** [comment]
There is an option (not automatic, default false) for nicehash mode in the pool settings, which may fix the other pool that is in nicehash mode.  Command line option `--nicehash`

**Comment by RCTORONTO** [comment]
so the cn-heavy mining is going well, I wanted to ask: is there any way to have xmrig/xmrig-cuda say which card found the accepted share when it prints "nvidia   accepted ......" - would that be functionality in the xmrig-cuda or in xmrig itself?  Does the Cuda backend distinguish which card adapter provided the accepted share, or does the backend simply group all the available devices into one sort of processor as the "backend"?

**Comment by Spudz76** [comment]
I don't think the GPU-Index is sent with the result to the main miner thread (from the plugin) so it can't know.  All it gets is "a result from the CUDA backend", double checks it on the CPU, and sends it if valid.

Might be something that could be wired, I haven't meddled with the plugin-api much though to know if it's possible for sure.

**Comment by RCTORONTO** [comment]
It would be a useful feature but I guess running cards singularly and seeing the hash rate will have to suffice :)

**Comment by DeeDeeRanged** [comment]
@Spudz76 Is there any difference between xmrig and xmrig-mo? Do I have to alter something in the json to make it, beside the usual like wallet+rig ?

**Comment by Spudz76** [comment]
@DeeDeeRanged There are only slight additions to the config most of it is identical and will work with either binary (but if the config is rewritten by the miner, it will strip the MO specific things).

**Comment by DeeDeeRanged** [comment]
@Spudz76 Thanks fyi.

**Comment by Spudz76** [comment]
Yes, I was going to mention I took all the fans off my HD7870's (mine were Tahiti Pro vs Pitcairn though) and put them on newer cards that had fan failures.  It was truly the best way they could help, haha.

The AMDGPU-Pro driver never really had full featured support for Sea Islands series and the support that was there essentially got abandoned but never removed.  PowerPlay and such do not work correctly at all, for example.  And then any of the other drivers don't support <GCN2.0 mostly.  They might work great with FGLRX however then you have to run Ubuntu 14 lol (fglrx doesn't work with newer kernels).  I had used fglrx with some Evergreen based GPU which is the tail end of TeraScale family (not even GCN1.0) and it worked somewhat well for what it was.  At some point it quit working, though, and maintaining an antique-software rig was far too difficult for less hashrate than a $5 CPU.

**Comment by DeeDeeRanged** [comment]
Well I have 1 rig with a 5800x cpu, GTX 1070 and a HD 7870 the GTX 1070 is mining eth and connected the HD 7870 to my 3440x1440 monitor as was quiet surprised how well it was still performing for card from 2013. The other rig has a 3600 cpu, rx 580 mining eth and hd 7870 hooked up to a FHD monitor. Using the cpuś to mine monero. It supplements my pension income and gives me something to do ;)
